libclc/ptx-nvidiacl/libspirv/synchronization/barrier.cl

```diff
@@ -9,14 +9,34 @@
 #include <spirv/spirv.h>
 #include <spirv/spirv_types.h>
 
+int __clc_nvvm_reflect_arch();
+
 _CLC_OVERLOAD _CLC_DEF void __spirv_MemoryBarrier(unsigned int memory,
                                                   unsigned int semantics) {
-  if (memory == CrossDevice) {
-    __nvvm_membar_sys();
-  } else if (memory == Device) {
-    __nvvm_membar_gl();
+
+  // for sm_70 and above membar becomes semantically identical to fence.sc.
+  // However sm_70 and above also introduces a lightweight fence.acq_rel that
+  // can be used to form either acquire or release strong operations.
+
+  unsigned int order = semantics & 0x1F;
+  if (order == None) {
```

Comment:
Why the empty `if-statement`?

Reply:
If I don't have the empty if statement at the beginning, then I would have to change the final else statement to e.g. check only for acquire/release/acq_rel etc. I thought this was the cleanest way.

Reply:
OK I can do !none

Reply:
Fixed this, thanks.